### PR TITLE
feat: search for attributes in flake locations

### PR DESCRIPTION
**Pull request by tomberek**
Will search for evalCatalog,packages,legacyPackgaes.

Should this be conditional on some sort of `{catalog = false;}`, but then the UX is assymetrical.

### CODE REVIEW

**Review by mkenigs — COMMENTED** [review]
- Just a thought, but I don't love that this makes resolution more ambiguous. Maybe not something we need to fix right now though
- I think this will result in putting an entry in the wrong place in catalog.json. If we fall back to using `getFlakeFlakePaths`, we should also be using `getFlakeCatalogPath`. That might require more of a refactor.

**Comment by aakropotkin** [comment]
This is related to RFC 0005 resolver which implements the search/resolution by catalog ID mentioned here.

This ticket doesn't necessarily need to be blocked by `resolver` integration if this is a quick fix; but I wanted to highlight this to avoid redundant efforts.

**Comment by mkenigs** [comment]
I'm going to close this for now as my understanding is that it's not currently blocking anything (correct me if I'm wrong @aakropotkin @limeytexan), catalogs will be moving to a separate branch so we'll need to update this anyways, and in it's current format this PR outputs incorrect data in catalog.json. We can reopen if necessary after we work out some more details of storing the catalog on a separate branch.

**Comment by limeytexan** [comment]
Hey @mkenigs, just met with @aakropotkin and he's struggling with a failing unit test that would be addressed with this patch. I know you and I had discussed this at length and concluded a) this patch would provide a workaround in the short term while we worked to implement the publish RFC, but b) this patch will not be required and in fact would be better removed in the long term once we get catalog and flake derivations split out into separate commit histories.

So with that I'd ask that we go ahead and merge this now to unblock Alex, and then come back to revert it later once the publish RFC has been fully implemented. Please let me know your thoughts, thanks.

File: modules/common.nix [hunk 1]
```diff
@@ -179,7 +179,7 @@ in {
         ++ packageAttrPath
         ++ version
       )
-    ];
+    ] ++ getFlakeFlakePaths packageAttrPath {};
```
Comment (mkenigs) [inline modules/common.nix:182]:
```suggestion
    # this is a temporary hack that leads to incorrect contents in catalog.json
    ] ++ getFlakeFlakePaths packageAttrPath {};
```

**Comment by mkenigs** [comment]
Okay merged to unblock for now, let's just make sure to revert at some point!

**Pull request merged**